define database model in mermaid code

Siddharta314 opened the pull request.

Siddharta314 committed:
define database model in mermaid code

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -0,0 +1,61 @@
+        String name
+        String googleID
+        String address
+        ObjectId[] dogs
```
what happend if i want to add a cat? Better use a field userId in other entity to relation it

Siddharta314 replied:
this issue is solved by using PET as general, or you want to create a separate PET table and connect the user with their userId?

BrianValdiviesoDev replied:
i prefer to use userId inside PET entity and delete "pets" field in users

Siddharta314 replied:
done

BrianValdiviesoDev commented on doc/database-model.md:
````diff
@@ -0,0 +1,61 @@
+
+```mermaid
+erDiagram
+    USER {
````
add email and password for login without social networks

Siddharta314 replied:
done

BrianValdiviesoDev commented on doc/database-model.md:
````diff
@@ -0,0 +1,61 @@
+
+```mermaid
+erDiagram
+    USER {
````
what happend with dates? i want to know when user starts to use the app

Siddharta314 replied:
timestamps are automatically created if set true

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -0,0 +1,61 @@
+        String address
+        ObjectId[] dogs
+    }
+    DOG {
```
why only dog? what do you think about calling it PET? if you add a "type" field you can use this entity to store dogs, cats, birds...

Siddharta314 replied:
i can change it to PET, though the specification for this app is exclusive por dogs

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -0,0 +1,61 @@
+        ObjectId[] photos
+    }
+    
+    MEDICAL_HISTORY {
```
i think this is better to store inside PET entity, anyway, i am missing a relation with the pet

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -0,0 +1,61 @@
+        String vaccine
+        String vet
+    }
+    APPOINTMENT {
```
how can i related it whit pet and user?

Siddharta314 replied:
user has a list of object pets

BrianValdiviesoDev commented on doc/database-model.md:
````diff
@@ -0,0 +1,61 @@
+
+```mermaid
+erDiagram
+    USER {
````
From business a user avatar is a requirement

Siddharta314 replied:
good :+1:

BrianValdiviesoDev commented on doc/database-model.md:
````diff
@@ -0,0 +1,61 @@
+# Database Entity Relationship Diagram
+
+```mermaid
+erDiagram
````
we need a role field and a role entity

Siddharta314 replied:
yes

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -0,0 +1,61 @@
+        String vet
+        String reason
+    }
+    PHOTO {
```
Why this entity?

Siddharta314 replied:
to upload multiple photos of the pet or pet related that aren't documents or medical history, if not necessary i can remove it

BrianValdiviesoDev replied:
if we need to upload a image for pets it's a better solution add a field image in pets which would be an array of strings or objects

Siddharta314 replied:
perfect. thanks

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -0,0 +1,61 @@
+        String[] allergies
+        String[] illness
+        ObjectId[] foods
+        ObjectId[] medical_history
```
if you are using camel case, use it in all fields

Siddharta314 replied:
then the table will be called medicalhistory. if there's a better name for it or only medical i'm open to suggestions

BrianValdiviesoDev replied:
that's ok

Siddharta314 committed:
issues resolved for PR :
 - added to user avatarURL, description, and role
 - created entity ROLE and changed some names like DOG to PET

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -6,11 +6,20 @@ erDiagram
         String address
-        ObjectId[] dogs
+        String role
+        ObjectId[] pets
```
delete this field and add userId in PET

Siddharta314 replied:
done

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -6,11 +6,20 @@ erDiagram
     }
-    DOG {
+
+    PET {
```
add userId to relation it with users

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -25,17 +34,18 @@ erDiagram
         ObjectId[] foods
-        ObjectId[] medical_history
+        ObjectId[] medicalHistory
         ObjectId[] appointments
```
change it by "events"

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -25,17 +34,18 @@ erDiagram
         String vet
     }
+
     APPOINTMENT {
```
change it by "EVENT" with fields name, date, petId, description

BrianValdiviesoDev commented on doc/database-model.md:
```diff
@@ -0,0 +1,72 @@
+        String url
+        String description
+    }
+    FOOD {
```
delete this entity, we will use "events" for this purpose

Siddharta314 replied:
done

Siddharta314 committed:
PET connected to user with userId, other small changes for PR
